File: 111-Minimum-Depth-of-Binary-Tree/solve.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        
        if (root == NULL)
            return 0;
        
        if(root->left == NULL && root->right == NULL){
            return 1;
        }
        
        int leftDepth, rightDepth;
        
        if (root->left == NULL)
            return minDepth(root->right) + 1;
            
        if (root->right == NULL)
            return minDepth(root->left) + 1;
        
     
        leftDepth = minDepth(root->left);
        rightDepth = minDepth(root->right);
        if (leftDepth <= rightDepth)    
            return leftDepth + 1;
        else
            return rightDepth + 1;
    
        
        
            
        
    }
};

File: 101-Symmetric-Tree/solve.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public boolean isSymmetric(TreeNode root) {
        
        if (root == null)
            return true;
        if (root.left != null && root.right == null)
            return false;
        if (root.left == null && root.right != null)
            return false;
            
        return isSame(root.left, root.right);
        
    }
    
    public boolean isSame(TreeNode left, TreeNode right){
        
        if (left == null && right == null)
            return true;
        if (left == null && right != null)
            return false;
        if (left != null && right == null)
            return false;
        if (left.val != right.val)
            return false;
        
        boolean one = isSame(left.right, right.left);
        if (one != true)
            return false;
        
        boolean two = isSame(left.left, right.right);
        if (two != true)
            return false;
        
        return true;
        
    }
}
